src/hooks/use-dialog/index.tsx

```diff
@@ -0,0 +1,42 @@
+import { FC, useMemo, useState } from 'react'
+
+import { Dialog, DialogProps } from '../../components/dialog'
+
+type RestrictedDialogComponentProps = Partial<DialogProps> & { footerItems: DialogProps['footerItems'] }
+
+export type UseDialog = [
+  Dialog: FC<RestrictedDialogComponentProps>,
+  closeDialog: () => void,
+  openDialog: () => void,
+  dialogIsOpen: boolean,
+]
+
+/**
+ * A hook for creating a dialog component.
+ * This hook have an isolated state that already handle to close and open the dialog.
+ *
+ * The returned type for this hook is a tuple
+ * 1. The dialog component
+ * 2. A function to close the dialog
+ * 3. A function to open the dialog
+ * 4. A boolean indicating whether the dialog is open
+ */
```

Comment:
**suggestion:** Personally, I think this hook is doing too much. Specifically, I don't think it should have any UI concerns baked into it (namely the creation of the `DialogComponent`). Dynamically-created React components prevent us from optimising the re-render performance of our UIs.

Personally, I would drop this `useDialog` hook entirely. It's not hard for consumers to manage a local boolean state value themselves for now; what consumers really need from a hook like this are all the a11y props that are needed to communicate that a particular trigger element controls a particular dialog and implementing that would be best left to a separate ticket.

Reply:
agree, that would also reduce the LoC and files changed; would be more better if it's on a ticket